optimum/habana/diffusers/models/unet_2d_condition.py

```diff
@@ -1,349 +1,352 @@
-    #   [batch,                    1, key_tokens]
-    # this helps to broadcast it as a bias over attention scores, which will be in one of the following shapes:
-    #   [batch,  heads, query_tokens, key_tokens] (e.g. torch sdp attn)
-    #   [batch * heads, query_tokens, key_tokens] (e.g. xformers or classic attn)
-    if attention_mask is not None:
-        # assume that mask is expressed as:
-        #   (1 = keep,      0 = discard)
-        # convert mask into a bias that can be added to attention scores:
-        #       (keep = +0,     discard = -10000.0)
-        attention_mask = (1 - attention_mask.to(sample.dtype)) * -10000.0
-        attention_mask = attention_mask.unsqueeze(1)
-
-    # convert encoder_attention_mask to a bias the same way we do for attention_mask
-    if encoder_attention_mask is not None:
-        encoder_attention_mask = (1 - encoder_attention_mask.to(sample.dtype)) * -10000.0
-        encoder_attention_mask = encoder_attention_mask.unsqueeze(1)
-
-    # 0. center input if necessary
-    if self.config.center_input_sample:
-        sample = 2 * sample - 1.0
-
-    # 1. time
-    timesteps = timestep
-    if not torch.is_tensor(timesteps):
-        # TODO: this requires sync between CPU and GPU. So try to pass timesteps as tensors if you can
-        # This would be a good case for the `match` statement (Python 3.10+)
-        is_mps = sample.device.type == "mps"
-        if isinstance(timestep, float):
-            dtype = torch.float32 if is_mps else torch.float64
-        else:
-            dtype = torch.int32 if is_mps else torch.int64
-        timesteps = torch.tensor([timesteps], dtype=dtype, device=sample.device)
-    elif len(timesteps.shape) == 0:
-        timesteps = timesteps[None].to(sample.device)
-
-    # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
-    timesteps = timesteps.expand(sample.shape[0])
-
-    t_emb = self.time_proj(timesteps)
-
-    # `Timesteps` does not contain any weights and will always return f32 tensors
-    # but time_embedding might actually be running in fp16. so we need to cast here.
-    # there might be better ways to encapsulate this.
-    t_emb = t_emb.to(dtype=sample.dtype)
-
-    emb = self.time_embedding(t_emb, timestep_cond)
-    aug_emb = None
-
-    if self.class_embedding is not None:
-        if class_labels is None:
-            raise ValueError("class_labels should be provided when num_class_embeds > 0")
-
-        if self.config.class_embed_type == "timestep":
-            class_labels = self.time_proj(class_labels)
-
-            # `Timesteps` does not contain any weights and will always return f32 tensors
-            # there might be better ways to encapsulate this.
-            class_labels = class_labels.to(dtype=sample.dtype)
-
-        class_emb = self.class_embedding(class_labels).to(dtype=sample.dtype)
-
-        if self.config.class_embeddings_concat:
-            emb = torch.cat([emb, class_emb], dim=-1)
+        self,
```

Comment:
Can you, please, fix indentation?   (remove 4 spaces from all lines in this function)